climada/hazard/base.py

```diff
@@ -1393,3 +1442,204 @@ def _read_att_excel(self, file_name, var_names):
+            )
+
+
+def join_centroids(haz_list):
+    """
+    Find centroids comon to all.
+
+    Centroids of all hazards must either all be rasters with the same
+    resolution, or all be points.
+
+    The centroids of all hazards are combined together. If points,
+    all points are combined. If rasters, a global raster is defined.
+
+    Parameters
+    ----------
+    haz_list : list(climada.hazard.Hazard())
+        List of hazard with centroids to join
+
+    Returns
+    -------
+    centroids : climada.hazard.Centroids()
+        Centroids containing the centroids of all hazards.
+
+    """
+    centroids = copy.deepcopy(haz_list[0].centroids)
+    for haz in haz_list[1:]:
+        if not centroids.equal(haz.centroids):
+            centroids.append(haz.centroids)
+    centroids.remove_duplicate_points()
+    return centroids
+
+def concatenate_hazard(haz_list, centroids=None, threshold=100):
+    """
+    Concatenate events of several hazards of same type.
+
+    Centroids of all hazards must either all be rasters with the same
+    resolution, or all be points.
+
+    By default, the centroids of all hazards are combined together. If points,
+    all points are combined. If rasters, a global raster is defined.
+
+    If centroids is not None, all hazards are defined on these centroids.
+    Centoids of hazards in haz_list not in centroids are mapped onto the
+    nearest point. If a point is further than threshold from the closet
+    centroid, the concatenation fails.
+
+    Parameters
+    ----------
+    haz_list: list(climada.hazard.Hazard())
+        Hazard instances of the same hazard type
+    centtroids: climada.hazard.Centroids(), optional
+        Centroids instance on which to map the concatenated hazard.
+        Default is None.
+    threshold: int or float
+        Threshold for mapping centoids of hazards in haz_list not in centroids.
+        Argument is passed to climada.util.coordinates.assign_coordinates.
+        Default is 100.
+
+    Returns
+    -------
+    haz_concat: climada.hazard.Hazard()
+        Concatenated hazard.
+
+    Raises
+    ------
+    ValueError
+    """
+    #Check units consistency among hazards
+    units = {haz.units for haz in haz_list}
+    if len(units) > 1:
+        raise TypeError("The haz_list contains hazards with different"
+                        "units %f. The hazards are incompatible and"
+                        "cannot be concatenated.", units)
+
+    #Define comon centroids
+    if centroids is None:
+        centroids = join_centroids(haz_list)
+
+    haz_concat = Hazard()
+    haz_concat.units = haz_list[0].units
+    haz_concat.centroids = centroids
+
+    #Indices for mapping matrices onto common centroids
+    hazcent_in_cent_idx_list = [
+        u_coord.assign_coordinates(haz.centroids.coord, centroids.coord,
+                                   threshold=threshold)
+        for haz in haz_list
+        ]
+
+    if -1 in hazcent_in_cent_idx_list:
```

Comment:
isn't this a list of lists?
```suggestion
    if any([-1 in idx_list for idx_list in hazcent_in_cent_idx_list]):
```

Reply:
Well spotted, I went too fast. Thanks!